sw/device/silicon_creator/rom/e2e/boot_policy_rollback/BUILD

```diff
@@ -38,21 +39,26 @@ BOOT_POLICY_ROLLBACK_CASES = [
         "a": 0,
         "b": 0,
         "exit_success": MSG_TEMPLATE_BFV.format(hex_digits(CONST.BFV.BOOT_POLICY.ROLLBACK)),
+        # By default the failure message include BFV but this will conflict with what we're looking for.
+        "exit_failure": "((FAIL|FAULT).*\n)",
```

Comment:
I think we still want to capture other BFVs as failure modes here no?  Just not the one we are expecting. Since when the ROM fails to boot it will never print "FAIL", Otherwise will we be waiting for this test to timeout in failure mode.

Reply:
Hmm, the question is how to capture it then. Rust regex doesn't have negative lookahead.

Reply:
yeah... I was looking into that earlier too. I suppose the behavior we want is really a string comparison after the regex match. But for now its fine to just let this timeout. Maybe just drop a comment and file an issue so we can get this fix to unblock other work streams.

Reply:
Alternatively, you can build a regex to match `BFV: <the complement of the expected value>`. If the expected value is say `1234` then the regex would be `BFV: ([^1])|(1[^2])|12[^3]|123[^4]`. Since the ROM is expected to print the BFV is any case, this would not wait for a timeout I believe.

Reply:
We'll need to generate this in Starlark though. It's definitely possible it might look ugly.

Reply:
I think something like that would work (added to `e2e/defs.bzl`):
```python
def msg_template_bfv_all_except(bfv_const):
   hex_str = hex_digits(bfv_const)
   regex = [
        "({}[^{}])".format(
            hex_str[0..idx],
            digit,
        )
        for (idx,digit) in enumerate(hex_str.elems())
   ].join("|")
   return MSG_TEMPLATE_BFV.format(regex0
```
Usage:
```python
msg_template_bfv_all_except(CONST.BFV.BOOT_POLICY.ROLLBACK)
```
I can look into more details and debug the above code if you want to continue this way.

Reply:
Thanks Amaury. I'll have a try